sys/riotboot/hdr.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_riotboot_hdr
+ * @{
+ *
+ * @file
+ * @brief       RIOT header helpers and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stddef.h>
+#include <stdio.h>
+
+#ifdef RIOT_VERSION
```

Comment:
I don't understand why `log.h` is included with RIOT_VERSION is defined. I see no good reason for just having `#include "log.h"`. Maybe there is one ?

Reply:
The reason is that when this file is compiled without RIOT the `LOG_*` macros won't be available, unless log.h is included, but it only exists on the RIOT context. Thus, whenever it doesn't compile for RIOT, the `LOG_DEBUG` macro is defined as a simple `prinft()`.